geest/core/workflows/default_index_score_workflow.py

```diff
@@ -36,212 +35,110 @@ def __init__(
         """
-        if self.feedback.isCanceled():  # Check for cancellation before starting
-            QgsMessageLog.logMessage(
-                "Workflow canceled before creating VRT.",
-                tag="Geest",
-                level=Qgis.Warning,
-            )
-            return
-
-        if output_vrt_name is None:
-            output_vrt_name = f"{self.layer_id}.vrt"
 
-        QgsMessageLog.logMessage(
-            f"Creating VRT of rasters '{output_vrt_name}' layer to the map.",
-            tag="Geest",
-            level=Qgis.Info,
+        shapefile_path = os.path.join(self.workflow_directory, f"{output_prefix}.shp")
+        scored_layer = QgsVectorLayer(
+            shapefile_path,
+            "scored_boundary_layer",
+            "ogr",
         )
-        # Directory containing rasters
-        raster_dir = os.path.dirname(output_vrt_name)
-        raster_files = glob.glob(os.path.join(raster_dir, "*.tif"))
+        if not scored_layer.isValid():
+            raise QgsProcessingException(f"Failed to load shapefile: {shapefile_path}")
 
-        if not raster_files:
-            QgsMessageLog.logMessage(
-                "No rasters found to combine into VRT.",
-                tag="Geest",
-                level=Qgis.Warning,
+        # Start editing mode
+        scored_layer.startEditing()
+
+        # Check if the "score" field already exists
+        if scored_layer.fields().indexFromName("score") == -1:
+            # Add the "score" field if it doesn't exist
+            scored_layer.dataProvider().addAttributes(
+                [QgsField("score", QVariant.Double)]
             )
-            return
-
-        vrt_filepath = os.path.join(raster_dir, output_vrt_name)
-
-        # Define the VRT parameters
-        params = {
-            "INPUT": raster_files,
-            "RESOLUTION": 0,  # Use highest resolution among input files
-            "SEPARATE": False,  # Combine all input rasters as a single band
-            "OUTPUT": vrt_filepath,
-            "PROJ_DIFFERENCE": False,
-            "ADD_ALPHA": False,
-            "ASSIGN_CRS": None,
-            "RESAMPLING": 0,
-            "SRC_NODATA": "255",
-            "EXTRA": "",
-        }
-
-        # Run the gdal:buildvrt processing algorithm to create the VRT
-        processing.run("gdal:buildvirtualraster", params)
-        QgsMessageLog.logMessage(
-            f"Created VRT: {vrt_filepath}", tag="Geest", level=Qgis.Info
-        )
+            scored_layer.updateFields()  # Make sure the layer is aware of the new field
 
-        # Add the VRT to the QGIS map
-        vrt_layer = QgsRasterLayer(vrt_filepath, f"{self.layer_id}")
+        # Get the index of the "score" field
+        score_field_index = scored_layer.fields().indexFromName("score")
 
-        if not vrt_layer.isValid():
+        # Update each feature with the score value
+        for feature in scored_layer.getFeatures():
```

Comment:
I don't think we need this loop since the iterator already passes you the feature to be scored